ReactWindows/ReactNative/Views/Text/ReactTextShadowNode.cs

```diff
@@ -198,6 +212,66 @@ public void SetFontStyle(string fontStyleString)
         }
 
         /// <summary>
+        /// Sets the text alignment.
+        /// </summary>
+        /// <param name="textAlign">The text alignment string.</param>
+        [ReactProp(ViewProps.TextAlign)]
+        public void SetTextAlign(string textAlign)
+        {
+            if (textAlign != null)
+            {
+                if (textAlign.Equals("auto"))
+                {
+                    _textAlignment = TextAlignment.DetectFromContent;
+                }
+                else
+                {
+                    _textAlignment = EnumHelpers.Parse<TextAlignment>(textAlign);
+                }
+                MarkUpdated();
+            }
+        }
+
+        /// <summary>
+        /// Sets the text decoration.
+        /// </summary>
+        /// <param name="textDecoration">The text decoration string.</param>
+        [ReactProp(ViewProps.TextDecorationLine)]
+        public void SetTextDecoration(string textDecoration)
+        {
+            if (textDecoration != null)
+            {
+                if (textDecoration.Contains("underline"))
+                {
+                    _underline = true;
+                    MarkUpdated();
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the maximum number of lines.
+        /// </summary>
+        /// <param name="numberOfLines">Max number of lines.</param>
+        [ReactProp(ViewProps.NumberOfLines)]
+        public virtual void SetNumberOfLines(int numberOfLines)
+        {
+            _numberOfLines = numberOfLines;
+            MarkUpdated();
+        }
+
+        /// <summary>
+        /// Sets the line height.
+        /// </summary>
+        /// <param name="lineHeight">The line height.</param>
+        [ReactProp(ViewProps.LineHeight)]
+        public virtual void SetLineHeight(double lineHeight)
```

Comment:
Just a question, should these be overridden in the ReactRawTextShadowNode and ReactVirtualTextShadowNode classes to throw (since they can only be set on the outermost text block)?  Or is this handled in JavaScript?